core/templates/dev/head/pages/library/Library.js

```diff
@@ -80,6 +80,37 @@ oppia.controller('Library', [
       $http.get('/libraryindexhandler').success(function(data) {
         $scope.libraryGroups = data.activity_summary_dicts_by_category;
 
+        for (var i = 0; i < $scope.libraryGroups.length; i++) {
+          var categoryActivity = $scope.libraryGroups[i].activity_summary_dicts;
+          for (var j = 0; j < categoryActivity.length; j++) {
+            activityType = categoryActivity[j].activity_type;
+            activityId = categoryActivity[j].id;
+
+            var learnerPlaylistUrl = (
+              UrlInterpolationService.interpolateUrl(
+                '/learnerplaylistactivityhandler/<activityType>/<activityId>', {
+                  activityType: activityType,
+                  activityId: activityId
+                }));
+            $http.post(learnerPlaylistUrl, {})
```

Comment:
This is still N $http.post() requests, where N is the number of activities. This needs to be addressed prior to submit.

As explained in previous comments, we should modify the backend so that all the relevant data is available in the single fetch for library data that the library page makes at the outset.

Reply:
Okay, I missed that point. I've implemented a different approch and made a commit, PTAL